pkg/discovery/servicehandler.go

```diff
@@ -322,48 +319,16 @@ func getExchangeAssetSpecFile(exchangeFiles []anypoint.ExchangeFile) *anypoint.E
 	return &exchangeFiles[0]
 }
 
-// specYAMLToJSON - if the spec is yaml convert it to json, SDK doesn't handle yaml.
-func specYAMLToJSON(specContent []byte) []byte {
-	specMap := make(map[string]interface{})
-	// check if the content is already json
-	err := json.Unmarshal(specContent, &specMap)
-	if err == nil {
-		return specContent
-	}
-
-	// check if the content is already yaml
-	err = yaml.Unmarshal(specContent, &specMap)
-	if err != nil {
-		// Not yaml, nothing more to be done
-		return specContent
-	}
-
-	bts, err := yaml.YAMLToJSON(specContent)
-	if err != nil {
-		return specContent
-	}
-	return bts
-}
-
-// getSpecType determines the correct resource type for the asset.
-func getSpecType(file *anypoint.ExchangeFile, specContent []byte) (string, error) {
-	if file.Classifier == apic.Wsdl {
-		return apic.Wsdl, nil
-	}
-
-	if specContent != nil {
-		jsonMap := make(map[string]interface{})
-		err := json.Unmarshal(specContent, &jsonMap)
-		if err != nil {
-			return "", err
-		}
-		if _, isSwagger := jsonMap["swagger"]; isSwagger {
-			return apic.Oas2, nil
-		} else if _, isOpenAPI := jsonMap["openapi"]; isOpenAPI {
-			return apic.Oas3, nil
+func getExchangeAssetWithRamlSpecFile(exchangeFiles []anypoint.ExchangeFile) *anypoint.ExchangeFile {
+	for i := range exchangeFiles {
+		c := exchangeFiles[i].Classifier
+		if (c == "oas" || c == "fat-oas" || c == "raml" || c == "fat-raml" || c == "wsdl") &&
```

Comment:
you can probably use the specPreference map for this conditional

`
if _, found := specPreference[c]; found {
....
}
`